--- ./arch/arm/dts/fsl-imx8mm-evk.dts
+++ ./arch/arm/dts/fsl-imx8mm-evk.dts
@@ -17,7 +17,7 @@
 #include "fsl-imx8mm.dtsi"
 
 / {
-	model = "FSL i.MX8MM EVK board";
+	model = "FSL i.MX8MM Kraken board";
 	compatible = "fsl,imx8mm-evk", "fsl,imx8mm";
 
 	chosen {
@@ -400,42 +400,6 @@
 	scl-gpios = <&gpio5 16 GPIO_ACTIVE_HIGH>;
 	sda-gpios = <&gpio5 17 GPIO_ACTIVE_HIGH>;
 	status = "okay";
-
-	typec_ptn5110_1: ptn5110@50 {
-		compatible = "usb,tcpci";
-		reg = <0x50>;
-		src-pdos = <0x380190c8>;
-		snk-pdos = <0x380190c8 0x3802d0c8>;
-		max-snk-mv = <9000>;
-		max-snk-ma = <2000>;
-		op-snk-mw = <9000>;
-		max-snk-mw = <18000>;
-		port-type = "drp";
-		default-role = "sink";
-	};
-
-	typec_ptn5110_2: ptn5110@52 {
-		compatible = "usb,tcpci";
-		reg = <0x52>;
-		src-pdos = <0x380190c8>;
-		snk-pdos = <0x380190c8 0x3802d0c8>;
-		max-snk-mv = <9000>;
-		max-snk-ma = <2000>;
-		op-snk-mw = <9000>;
-		max-snk-mw = <18000>;
-		port-type = "drp";
-		default-role = "sink";
-	};
-};
-
-&i2c3 {
-	clock-frequency = <100000>;
-	pinctrl-names = "default", "gpio";
-	pinctrl-0 = <&pinctrl_i2c3>;
-	pinctrl-1 = <&pinctrl_i2c3_gpio>;
-	scl-gpios = <&gpio5 18 GPIO_ACTIVE_HIGH>;
-	sda-gpios = <&gpio5 19 GPIO_ACTIVE_HIGH>;
-	status = "okay";
 };
 
 &flexspi0 {
@@ -465,12 +429,9 @@
 		#address-cells = <1>;
 		#size-cells = <0>;
 
-		ethphy0: ethernet-phy@0 {
+		ethphy0: ethernet-phy@5 {
 			compatible = "ethernet-phy-ieee802.3-c22";
-			reg = <0>;
-			at803x,led-act-blind-workaround;
-			at803x,eee-okay;
-			at803x,vddio-1p8v;
+			reg = <5>;
 		};
 	};
 };
@@ -514,11 +474,11 @@
 };
 
 &usbotg1 {
+	dr_mode = "peripheral";
 	status = "okay";
-	extcon = <&typec_ptn5110_1>;
 };
 
 &usbotg2 {
+	dr_mode = "peripheral";
 	status = "okay";
-	extcon = <&typec_ptn5110_2>;
 };
