--- ./arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
+++ ./arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
@@ -25,31 +25,55 @@
 		stdout-path = &uart2;
 	};
 
-	leds {
-		compatible = "gpio-leds";
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_gpio_led>;
+	pwm_leds {
+		compatible = "pwm-leds";
 
-		status {
-			label = "status";
-			gpios = <&gpio3 16 0>;
-			default-state = "on";
+		red {
+			label = "LED_RED";
+			pwms = <&pwm1 0 50000>;
+			linux,default-trigger = "none";
+			max-brightness = <255>;
+		};
+		green {
+			label = "LED_GREEN";
+			pwms = <&pwm2 0 50000>;
+			// default booting pattern
+			linux,default-trigger = "timer";
+			led-pattern = "500,500";
+			max-brightness = <255>;
+		};
+		blue {
+			label = "LED_BLUE";
+			pwms = <&pwm3 0 50000>;
+			linux,default-trigger = "none";
+			max-brightness = <255>;
 		};
 	};
 
-	modem_reset: modem-reset {
-		compatible = "gpio-reset";
-		reset-gpios = <&gpio2 6 GPIO_ACTIVE_LOW>;
-		reset-delay-us = <2000>;
-		reset-post-delay-ms = <40>;
-		#reset-cells = <0>;
-	};
-
-	ir_recv: ir-receiver {
-		compatible = "gpio-ir-receiver";
-		gpios = <&gpio1 13 GPIO_ACTIVE_LOW>;
+	gpio_keys {
+		compatible = "gpio-keys";
 		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_ir_recv>;
+		pinctrl-0 = <&pinctrl_gpio_keys>;
+
+		switch1 {
+			label = "SW1";
+			linux,code = <59>; /* Emit <F1> key code */
+			gpios = <&gpio5 2 GPIO_ACTIVE_LOW>;
+			interrupt-parent = <&gpio5>;
+			interrupts = <2 IRQ_TYPE_EDGE_FALLING>; /* GPIO #2. High-to-low edge triggered */
+			debounce-interval = <20>;
+			gpio-key,wakeup = <1>;
+		};
+
+		switch2 {
+			label = "SW2";
+			linux,code = <60>; /* Emit <F2> key code */
+			gpios = <&gpio4 31 GPIO_ACTIVE_LOW>;
+			interrupt-parent = <&gpio4>;
+			interrupts = <31 IRQ_TYPE_EDGE_FALLING>; /* GPIO #31. High-to-low edge triggered */
+			debounce-interval = <20>;
+			gpio-key,wakeup = <1>;
+		};
 	};
 
 	regulators {
@@ -77,114 +101,29 @@
 			off-on-delay = <20000>;
 			enable-active-high;
 		};
-
-		reg_audio_board: regulator-audio-board {
-			compatible = "regulator-fixed";
-			regulator-name = "EXT_PWREN";
-			regulator-min-microvolt = <3300000>;
-			regulator-max-microvolt = <3300000>;
-			enable-active-high;
-			startup-delay-us = <300000>;
-			gpio = <&pca6416 1 GPIO_ACTIVE_HIGH>;
-		};
-	};
-
-	wm8524: wm8524 {
-		compatible = "wlf,wm8524";
-		clocks = <&clk IMX8MM_CLK_SAI3_ROOT>;
-		clock-names = "mclk";
-		wlf,mute-gpios = <&gpio5 21 GPIO_ACTIVE_LOW>;
-	};
-
-	sound-wm8524 {
-		compatible = "fsl,imx-audio-wm8524";
-		model = "wm8524-audio";
-		audio-cpu = <&sai3>;
-		audio-codec = <&wm8524>;
-		audio-routing =
-			"Line Out Jack", "LINEVOUTL",
-			"Line Out Jack", "LINEVOUTR";
-	};
-
-	sound-ak4458 {
-		compatible = "fsl,imx-audio-ak4458";
-		model = "ak4458-audio";
-		audio-cpu = <&sai1>;
-		audio-codec = <&ak4458_1>, <&ak4458_2>;
-		ak4458,pdn-gpio = <&pca6416 4 GPIO_ACTIVE_HIGH>;
-	};
-
-	sound-ak5558 {
-		compatible = "fsl,imx-audio-ak5558";
-		model = "ak5558-audio";
-		audio-cpu = <&sai5>;
-		audio-codec = <&ak5558>;
-		status = "disabled";
-	};
-
-	sound-ak4497 {
-		compatible = "fsl,imx-audio-ak4497";
-		model = "ak4497-audio";
-		audio-cpu = <&sai1>;
-		audio-codec = <&ak4497>;
-		status = "disabled";
-	};
-
-	sound-spdif {
-		compatible = "fsl,imx-audio-spdif";
-		model = "imx-spdif";
-		spdif-controller = <&spdif1>;
-		spdif-out;
-		spdif-in;
-	};
-
-	sound-micfil {
-		compatible = "fsl,imx-audio-micfil";
-		model = "imx-audio-micfil";
-		cpu-dai = <&micfil>;
 	};
 };
 
-&clk {
-	assigned-clocks = <&clk IMX8MM_AUDIO_PLL1>, <&clk IMX8MM_AUDIO_PLL2>;
-	assigned-clock-rates = <786432000>, <722534400>;
+&pwm1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pwm1>;
+	status = "okay";
+};
+&pwm2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pwm2>;
+	status = "okay";
+};
+&pwm3 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pwm3>;
+	status = "okay";
 };
 
 &iomuxc {
 	pinctrl-names = "default";
 
 	imx8mm-evk {
-		pinctrl_csi_pwn: csi_pwn_grp {
-			fsl,pins = <
-				MX8MM_IOMUXC_GPIO1_IO07_GPIO1_IO7		0x19
-			>;
-		};
-
-		pinctrl_ir_recv: ir_recv {
-			fsl,pins = <
-				MX8MM_IOMUXC_GPIO1_IO13_GPIO1_IO13		0x14f
-			>;
-		};
-
-		pinctrl_csi_rst: csi_rst_grp {
-			fsl,pins = <
-				MX8MM_IOMUXC_GPIO1_IO06_GPIO1_IO6		0x19
-				MX8MM_IOMUXC_GPIO1_IO14_CCMSRCGPCMIX_CLKO1	0x59
-			>;
-		};
-
-		pinctrl_mipi_dsi_en: mipi_dsi_en {
-			fsl,pins = <
-				MX8MM_IOMUXC_GPIO1_IO08_GPIO1_IO8		0x16
-			>;
-		};
-
-		pinctrl_i2c2_synaptics_dsx_io: synaptics_dsx_iogrp {
-			fsl,pins = <
-				MX8MM_IOMUXC_GPIO1_IO09_GPIO1_IO9		0x19	/* Touch int */
-			>;
-		};
-
 		pinctrl_fec1: fec1grp {
 			fsl,pins = <
 				MX8MM_IOMUXC_ENET_MDC_ENET1_MDC		0x3
@@ -216,131 +155,60 @@
 			>;
 		};
 
-		pinctrl_gpio_led: gpioledgrp {
-			fsl,pins = <
-				MX8MM_IOMUXC_NAND_READY_B_GPIO3_IO16	0x19
-			>;
-		};
-
-		pinctrl_i2c1: i2c1grp {
-			fsl,pins = <
-				MX8MM_IOMUXC_I2C1_SCL_I2C1_SCL			0x400001c3
-				MX8MM_IOMUXC_I2C1_SDA_I2C1_SDA			0x400001c3
-			>;
-		};
-
-		pinctrl_i2c2: i2c2grp {
-			fsl,pins = <
-				MX8MM_IOMUXC_I2C2_SCL_I2C2_SCL			0x400001c3
-				MX8MM_IOMUXC_I2C2_SDA_I2C2_SDA			0x400001c3
-			>;
-		};
-
-		pinctrl_i2c3: i2c3grp {
-			fsl,pins = <
-				MX8MM_IOMUXC_I2C3_SCL_I2C3_SCL			0x400001c3
-				MX8MM_IOMUXC_I2C3_SDA_I2C3_SDA			0x400001c3
-			>;
-		};
-
-		pinctrl_pcie0: pcie0grp {
-			fsl,pins = <
-				MX8MM_IOMUXC_I2C4_SCL_PCIE1_CLKREQ_B	0x61 /* open drain, pull up */
-				MX8MM_IOMUXC_GPIO1_IO05_GPIO1_IO5	0x41
-				MX8MM_IOMUXC_SAI2_RXFS_GPIO4_IO21	0x41
-			>;
-		};
-
-		pinctrl_pmic: pmicirq {
+		pinctrl_pwm1: pwm1grp {
 			fsl,pins = <
-				MX8MM_IOMUXC_GPIO1_IO03_GPIO1_IO3		0x41
+				MX8MM_IOMUXC_GPIO1_IO01_PWM1_OUT	0x19
 			>;
 		};
 
-		pinctrl_typec1: typec1grp {
+		pinctrl_pwm2: pwm2grp {
 			fsl,pins = <
-				MX8MM_IOMUXC_SD1_STROBE_GPIO2_IO11	0x159
-			>;
+				MX8MM_IOMUXC_GPIO1_IO13_PWM2_OUT	0x19
+							>;
 		};
 
-		pinctrl_typec2: typec2grp {
+		pinctrl_pwm3: pwm3grp {
 			fsl,pins = <
-				MX8MM_IOMUXC_SD2_CD_B_GPIO2_IO12	0x159
+				MX8MM_IOMUXC_GPIO1_IO14_PWM3_OUT	0x19
 			>;
 		};
 
-		pinctrl_sai1: sai1grp {
+		pinctrl_gpio_keys: gpiokeysgrp {
 			fsl,pins = <
-				MX8MM_IOMUXC_SAI1_MCLK_SAI1_MCLK	0xd6
-				MX8MM_IOMUXC_SAI1_TXFS_SAI1_TX_SYNC	0xd6
-				MX8MM_IOMUXC_SAI1_RXD7_SAI1_TX_SYNC	0xd6
-				MX8MM_IOMUXC_SAI1_TXC_SAI1_TX_BCLK	0xd6
-				MX8MM_IOMUXC_SAI1_TXD0_SAI1_TX_DATA0	0xd6
-				MX8MM_IOMUXC_SAI1_TXD1_SAI1_TX_DATA1	0xd6
-				MX8MM_IOMUXC_SAI1_TXD2_SAI1_TX_DATA2	0xd6
-				MX8MM_IOMUXC_SAI1_TXD3_SAI1_TX_DATA3	0xd6
-				MX8MM_IOMUXC_SAI1_TXD4_SAI1_TX_DATA4	0xd6
-				MX8MM_IOMUXC_SAI1_TXD5_SAI1_TX_DATA5	0xd6
-				MX8MM_IOMUXC_SAI1_TXD6_SAI1_TX_DATA6	0xd6
-				MX8MM_IOMUXC_SAI1_TXD7_SAI1_TX_DATA7	0xd6
+				MX8MM_IOMUXC_SAI3_MCLK_GPIO5_IO2	0x159
+				MX8MM_IOMUXC_SAI3_TXFS_GPIO4_IO31	0x159
 			>;
 		};
 
-		pinctrl_sai1_dsd: sai1grp_dsd {
+		pinctrl_i2c1: i2c1grp {
 			fsl,pins = <
-				MX8MM_IOMUXC_SAI1_MCLK_SAI1_MCLK	0xd6
-				MX8MM_IOMUXC_SAI1_TXFS_SAI1_TX_SYNC	0xd6
-				MX8MM_IOMUXC_SAI1_RXD7_SAI1_TX_DATA4	0xd6
-				MX8MM_IOMUXC_SAI1_TXC_SAI1_TX_BCLK	0xd6
-				MX8MM_IOMUXC_SAI1_TXD0_SAI1_TX_DATA0	0xd6
-				MX8MM_IOMUXC_SAI1_TXD1_SAI1_TX_DATA1	0xd6
-				MX8MM_IOMUXC_SAI1_TXD2_SAI1_TX_DATA2	0xd6
-				MX8MM_IOMUXC_SAI1_TXD3_SAI1_TX_DATA3	0xd6
-				MX8MM_IOMUXC_SAI1_TXD4_SAI1_TX_DATA4	0xd6
-				MX8MM_IOMUXC_SAI1_TXD5_SAI1_TX_DATA5	0xd6
-				MX8MM_IOMUXC_SAI1_TXD6_SAI1_TX_DATA6	0xd6
-				MX8MM_IOMUXC_SAI1_TXD7_SAI1_TX_DATA7	0xd6
+				MX8MM_IOMUXC_I2C1_SCL_I2C1_SCL			0x400001c3
+				MX8MM_IOMUXC_I2C1_SDA_I2C1_SDA			0x400001c3
 			>;
 		};
 
-		pinctrl_sai3: sai3grp {
+		pinctrl_i2c2: i2c2grp {
 			fsl,pins = <
-				MX8MM_IOMUXC_SAI3_TXFS_SAI3_TX_SYNC     0xd6
-				MX8MM_IOMUXC_SAI3_TXC_SAI3_TX_BCLK      0xd6
-				MX8MM_IOMUXC_SAI3_MCLK_SAI3_MCLK        0xd6
-				MX8MM_IOMUXC_SAI3_TXD_SAI3_TX_DATA0     0xd6
-				MX8MM_IOMUXC_I2C4_SDA_GPIO5_IO21        0xd6
+				MX8MM_IOMUXC_I2C2_SCL_I2C2_SCL			0x400001c3
+				MX8MM_IOMUXC_I2C2_SDA_I2C2_SDA			0x400001c3
 			>;
 		};
-
-		pinctrl_sai5: sai5grp {
+		pinctrl_pcie0: pcie0grp {
 			fsl,pins = <
-				MX8MM_IOMUXC_SAI5_MCLK_SAI5_MCLK	0xd6
-				MX8MM_IOMUXC_SAI5_RXC_SAI5_RX_BCLK	0xd6
-				MX8MM_IOMUXC_SAI5_RXFS_SAI5_RX_SYNC	0xd6
-				MX8MM_IOMUXC_SAI5_RXD0_SAI5_RX_DATA0	0xd6
-				MX8MM_IOMUXC_SAI5_RXD1_SAI5_RX_DATA1    0xd6
-				MX8MM_IOMUXC_SAI5_RXD2_SAI5_RX_DATA2    0xd6
-				MX8MM_IOMUXC_SAI5_RXD3_SAI5_RX_DATA3    0xd6
+				MX8MM_IOMUXC_I2C4_SCL_PCIE1_CLKREQ_B	0x61 /* open drain, pull up */
+				MX8MM_IOMUXC_SAI1_MCLK_GPIO4_IO20	0x41
 			>;
 		};
 
-		pinctrl_pdm: pdmgrp {
+		pinctrl_pmic: pmicirq {
 			fsl,pins = <
-				MX8MM_IOMUXC_SAI5_MCLK_SAI5_MCLK	0xd6
-				MX8MM_IOMUXC_SAI5_RXC_PDM_CLK		0xd6
-				MX8MM_IOMUXC_SAI5_RXFS_SAI5_RX_SYNC	0xd6
-				MX8MM_IOMUXC_SAI5_RXD0_PDM_DATA0	0xd6
-				MX8MM_IOMUXC_SAI5_RXD1_PDM_DATA1	0xd6
-				MX8MM_IOMUXC_SAI5_RXD2_PDM_DATA2	0xd6
-				MX8MM_IOMUXC_SAI5_RXD3_PDM_DATA3	0xd6
+				MX8MM_IOMUXC_GPIO1_IO03_GPIO1_IO3		0x41
 			>;
 		};
 
-		pinctrl_spdif1: spdif1grp {
+		pinctrl_bqic: bqirq {
 			fsl,pins = <
-				MX8MM_IOMUXC_SPDIF_TX_SPDIF1_OUT	0xd6
-				MX8MM_IOMUXC_SPDIF_RX_SPDIF1_IN		0xd6
+				MX8MM_IOMUXC_GPIO1_IO00_GPIO1_IO0		0x20
 			>;
 		};
 
@@ -350,7 +218,8 @@
 				MX8MM_IOMUXC_UART1_TXD_UART1_DCE_TX	0x140
 				MX8MM_IOMUXC_UART3_RXD_UART1_DCE_CTS_B	0x140
 				MX8MM_IOMUXC_UART3_TXD_UART1_DCE_RTS_B	0x140
-				MX8MM_IOMUXC_SD1_DATA4_GPIO2_IO6	0x19
+				MX8MM_IOMUXC_SAI1_TXFS_GPIO4_IO10	0x159	/* RADIO1_#RESET pull-up */
+				MX8MM_IOMUXC_SAI5_MCLK_GPIO3_IO25	0x159	/* BTL_BUTTON1 pull-up */
 			>;
 		};
 
@@ -367,12 +236,16 @@
 				MX8MM_IOMUXC_ECSPI1_MOSI_UART3_DCE_TX		0x140
 				MX8MM_IOMUXC_ECSPI1_SS0_UART3_DCE_RTS_B		0x140
 				MX8MM_IOMUXC_ECSPI1_MISO_UART3_DCE_CTS_B	0x140
+				MX8MM_IOMUXC_SAI1_TXC_GPIO4_IO11	0x159	/* RADIO2_#RESET pull-up */
+				MX8MM_IOMUXC_SAI5_RXFS_GPIO3_IO19	0x159	/* BTL_BUTTON2 pull-up */
 			>;
 		};
 
-		pinctrl_usdhc1_gpio: usdhc1grpgpio {
+		pinctrl_uart4: uart4grp {
 			fsl,pins = <
-				MX8MM_IOMUXC_SD1_RESET_B_GPIO2_IO10	0x41
+				MX8MM_IOMUXC_UART4_RXD_UART4_DCE_RX		0x140
+				MX8MM_IOMUXC_UART4_TXD_UART4_DCE_TX		0x140
+				MX8MM_IOMUXC_SAI3_RXFS_GPIO4_IO28 		0x140 /* enable pullup. disable open drain, x1 drive strength */
 			>;
 		};
 
@@ -500,20 +373,43 @@
 			>;
 		};
 
+		pinctrl_ecspi2: ecspi2grp {
+			fsl,pins = <
+				MX8MM_IOMUXC_ECSPI2_SCLK_ECSPI2_SCLK		0x116
+				MX8MM_IOMUXC_ECSPI2_MOSI_ECSPI2_MOSI		0x116
+				MX8MM_IOMUXC_ECSPI2_MISO_ECSPI2_MISO		0x116
+				MX8MM_IOMUXC_ECSPI2_SS0_GPIO5_IO13			0x156
+			>;
+		};
+
+
+		pinctrl_wfx_irq: wfxirqgrp {
+			fsl,pins = <
+				MX8MM_IOMUXC_SD1_STROBE_GPIO2_IO11	0x41
+			>;
+		};
+
+		pinctrl_wfx_gpios: wfxgpiogrp {
+			fsl,pins = <
+				MX8MM_IOMUXC_SD1_RESET_B_GPIO2_IO10		0x41
+				MX8MM_IOMUXC_SAI5_RXD3_GPIO3_IO24		0x41
+			>;
+		};
+
 		pinctrl_wdog: wdoggrp {
 			fsl,pins = <
 				MX8MM_IOMUXC_GPIO1_IO02_WDOG1_WDOG_B		0xc6
 			>;
 		};
-	};
-};
 
-&csi1_bridge {
-	fsl,mipi-mode;
-	status = "okay";
-	port {
-		csi1_ep: endpoint {
-			remote-endpoint = <&csi1_mipi_ep>;
+		pinctrl_lte: ltegrp {
+			fsl,pins = <
+				MX8MM_IOMUXC_GPIO1_IO05_GPIO1_IO5		0x19	/* VBAT control	0 means on */
+				MX8MM_IOMUXC_GPIO1_IO06_GPIO1_IO6		0x19	/* PWRKEY 		1 means pressed */
+				MX8MM_IOMUXC_GPIO1_IO07_GPIO1_IO7		0x19	/* W_DISABLE 	1 means flight mode on */
+				MX8MM_IOMUXC_GPIO1_IO08_GPIO1_IO8		0x19	/* RESET		1 means reset */
+				MX8MM_IOMUXC_SAI3_RXC_GPIO4_IO29		0x19	/* USB BOOT		1 means USB boot */
+			>;
 		};
 	};
 };
@@ -681,225 +577,27 @@
 	pinctrl-0 = <&pinctrl_i2c2>;
 	status = "okay";
 
-	adv_bridge: adv7535@3d {
-		compatible = "adi,adv7533";
-		reg = <0x3d>;
-		adi,addr-cec = <0x3b>;
-		adi,dsi-lanes = <4>;
-		status = "okay";
-
-		port {
-			adv7535_from_dsim: endpoint {
-				remote-endpoint = <&dsim_to_adv7535>;
-			};
-		};
-	};
+	// https://elixir.bootlin.com/linux/v4.2/source/Documentation/devicetree/bindings/power/bq25890.txt
+	bq25890@6b {
+        compatible = "ti,bq25890";
+        reg = <0x6b>;
+		pinctrl-0 = <&pinctrl_bqic>;
+		interrupt-parent = <&gpio1>;
+		interrupts = <0 IRQ_TYPE_EDGE_FALLING>;
+		ti,bq25890_irq = <&gpio1 0>;
+
+        ti,battery-regulation-voltage = <4200000>;
+        ti,charge-current = <1000000>;
+        ti,termination-current = <50000>;
+        ti,precharge-current = <128000>;
+        ti,minimum-sys-voltage = <3600000>;
+        ti,boost-voltage = <4200000>;
+        ti,boost-max-current = <1000000>;
 
-	typec1_ptn5110: tcpci@50 {
-		compatible = "usb,tcpci";
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_typec1>;
-		reg = <0x50>;
-		interrupt-parent = <&gpio2>;
-		interrupts = <11 8>;
-		src-pdos = <0x380190c8>;
-		snk-pdos = <0x380190c8>;
-		/* Only can sink 5V for safe */
-		max-snk-mv = <5000>;
-		max-snk-ma = <3000>;
-		op-snk-mw = <10000>;
-		max-snk-mw = <15000>;
-		port-type = "drp";
-		default-role = "sink";
-		status = "okay";
+        ti,use-ilim-pin;
+        ti,thermal-regulation-threshold = <120>;
 	};
 
-	typec2_ptn5110: tcpci@52 {
-		compatible = "usb,tcpci";
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_typec2>;
-		reg = <0x52>;
-		interrupt-parent = <&gpio2>;
-		interrupts = <12 8>;
-		src-pdos = <0x380190c8>;
-		snk-pdos = <0x380190c8>;
-		/* Only can sink 5V for safe */
-		max-snk-mv = <5000>;
-		max-snk-ma = <3000>;
-		op-snk-mw = <10000>;
-		max-snk-mw = <15000>;
-		port-type = "drp";
-		default-role = "sink";
-		status = "okay";
-	};
-};
-
-
-&mipi_csi_1 {
-	#address-cells = <1>;
-	#size-cells = <0>;
-	status = "okay";
-	port {
-		mipi1_sensor_ep: endpoint1 {
-			remote-endpoint = <&ov5640_mipi1_ep>;
-			data-lanes = <2>;
-			csis-hs-settle = <13>;
-			csis-clk-settle = <2>;
-			csis-wclk;
-		};
-
-		csi1_mipi_ep: endpoint2 {
-			remote-endpoint = <&csi1_ep>;
-		};
-	};
-};
-
-&i2c3 {
-	clock-frequency = <100000>;
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_i2c3>;
-	status = "okay";
-
-	pca6416: gpio@20 {
-		compatible = "ti,tca6416";
-		reg = <0x20>;
-		gpio-controller;
-		#gpio-cells = <2>;
-	};
-
-	ak4458_1: ak4458@10 {
-		compatible = "asahi-kasei,ak4458";
-		reg = <0x10>;
-		AVDD-supply = <&reg_audio_board>;
-		DVDD-supply = <&reg_audio_board>;
-	};
-
-	ak4458_2: ak4458@12 {
-		compatible = "asahi-kasei,ak4458";
-		reg = <0x12>;
-		AVDD-supply = <&reg_audio_board>;
-		DVDD-supply = <&reg_audio_board>;
-	};
-
-	ak5558: ak5558@13 {
-		compatible = "asahi-kasei,ak5558";
-		reg = <0x13>;
-		ak5558,pdn-gpio = <&pca6416 3 GPIO_ACTIVE_HIGH>;
-		AVDD-supply = <&reg_audio_board>;
-		DVDD-supply = <&reg_audio_board>;
-	};
-
-	ak4497: ak4497@11 {
-		compatible = "asahi-kasei,ak4497";
-		reg = <0x11>;
-		ak4497,pdn-gpio = <&pca6416 5 GPIO_ACTIVE_HIGH>;
-		AVDD-supply = <&reg_audio_board>;
-		DVDD-supply = <&reg_audio_board>;
-	};
-
-	ov5640_mipi: ov5640_mipi@3c {
-		compatible = "ovti,ov5640_mipi";
-		reg = <0x3c>;
-		status = "okay";
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_csi_pwn>, <&pinctrl_csi_rst>;
-		clocks = <&clk IMX8MM_CLK_CLKO1_DIV>;
-		clock-names = "csi_mclk";
-		assigned-clocks = <&clk IMX8MM_CLK_CLKO1_SRC>,
-				  <&clk IMX8MM_CLK_CLKO1_DIV>;
-		assigned-clock-parents = <&clk IMX8MM_CLK_24M>;
-		assigned-clock-rates = <0>, <24000000>;
-		csi_id = <0>;
-		pwn-gpios = <&gpio1 7 GPIO_ACTIVE_HIGH>;
-		mclk = <24000000>;
-		mclk_source = <0>;
-		port {
-			ov5640_mipi1_ep: endpoint {
-				remote-endpoint = <&mipi1_sensor_ep>;
-			};
-		};
-	};
-};
-
-&lcdif {
-	status = "okay";
-};
-
-&mipi_dsi {
-	status = "okay";
-
-	port@1 {
-		dsim_to_adv7535: endpoint {
-			remote-endpoint = <&adv7535_from_dsim>;
-		};
-	};
-};
-
-&mu {
-	status = "okay";
-};
-
-&sai1 {
-	pinctrl-names = "default", "dsd";
-	pinctrl-0 = <&pinctrl_sai1>;
-	pinctrl-1 = <&pinctrl_sai1_dsd>;
-	assigned-clocks = <&clk IMX8MM_CLK_SAI1_SRC>,
-			<&clk IMX8MM_CLK_SAI1_DIV>;
-	assigned-clock-parents = <&clk IMX8MM_AUDIO_PLL1_OUT>;
-	assigned-clock-rates = <0>, <49152000>;
-	clocks = <&clk IMX8MM_CLK_SAI1_IPG>, <&clk IMX8MM_CLK_DUMMY>,
-		<&clk IMX8MM_CLK_SAI1_ROOT>, <&clk IMX8MM_CLK_DUMMY>,
-		<&clk IMX8MM_CLK_DUMMY>, <&clk IMX8MM_AUDIO_PLL1_OUT>,
-		<&clk IMX8MM_AUDIO_PLL2_OUT>;
-	clock-names = "bus", "mclk0", "mclk1", "mclk2", "mclk3", "pll8k", "pll11k";
-	fsl,sai-multi-lane;
-	fsl,dataline,dsd = <0 0xff 0xff 2 0xff 0x11>;
-	dmas = <&sdma2 0 26 0>, <&sdma2 1 26 0>;
-	status = "okay";
-};
-
-&sai3 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_sai3>;
-	assigned-clocks = <&clk IMX8MM_CLK_SAI3_SRC>,
-			<&clk IMX8MM_CLK_SAI3_DIV>;
-	assigned-clock-parents = <&clk IMX8MM_AUDIO_PLL1_OUT>;
-	assigned-clock-rates = <0>, <24576000>;
-	status = "okay";
-};
-
-&sai5 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_sai5>;
-	assigned-clocks = <&clk IMX8MM_CLK_SAI5_SRC>,
-			<&clk IMX8MM_CLK_SAI5_DIV>;
-	assigned-clock-parents = <&clk IMX8MM_AUDIO_PLL1_OUT>;
-	assigned-clock-rates = <0>, <49152000>;
-	clocks = <&clk IMX8MM_CLK_SAI5_IPG>, <&clk IMX8MM_CLK_DUMMY>,
-		<&clk IMX8MM_CLK_SAI5_ROOT>, <&clk IMX8MM_CLK_DUMMY>,
-		<&clk IMX8MM_CLK_DUMMY>, <&clk IMX8MM_AUDIO_PLL1_OUT>,
-		<&clk IMX8MM_AUDIO_PLL2_OUT>;
-	clock-names = "bus", "mclk0", "mclk1", "mclk2", "mclk3", "pll8k", "pll11k";
-	fsl,sai-asynchronous;
-	status = "disabled";
-};
-
-&spdif1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_spdif1>;
-	assigned-clocks = <&clk IMX8MM_CLK_SPDIF1_SRC>,
-			<&clk IMX8MM_CLK_SPDIF1_DIV>;
-	assigned-clock-parents = <&clk IMX8MM_AUDIO_PLL1_OUT>;
-	assigned-clock-rates = <0>, <24576000>;
-	clocks = <&clk IMX8MM_CLK_AUDIO_AHB_DIV>, <&clk IMX8MM_CLK_24M>,
-		<&clk IMX8MM_CLK_SPDIF1_DIV>, <&clk IMX8MM_CLK_DUMMY>,
-		<&clk IMX8MM_CLK_DUMMY>, <&clk IMX8MM_CLK_DUMMY>,
-		<&clk IMX8MM_CLK_AUDIO_AHB_DIV>, <&clk IMX8MM_CLK_DUMMY>,
-		<&clk IMX8MM_CLK_DUMMY>, <&clk IMX8MM_CLK_DUMMY>,
-		<&clk IMX8MM_AUDIO_PLL1_OUT>, <&clk IMX8MM_AUDIO_PLL2_OUT>;
-	clock-names = "core", "rxtx0", "rxtx1", "rxtx2", "rxtx3",
-		"rxtx4", "rxtx5", "rxtx6", "rxtx7", "spba", "pll8k", "pll11k";
-	status = "okay";
 };
 
 &fec1 {
@@ -914,12 +612,9 @@
 		#address-cells = <1>;
 		#size-cells = <0>;
 
-		ethphy0: ethernet-phy@0 {
+		ethphy0: ethernet-phy@5 {
 			compatible = "ethernet-phy-ieee802.3-c22";
-			reg = <0>;
-			at803x,led-act-blind-workaround;
-			at803x,eee-okay;
-			at803x,vddio-1p8v;
+			reg = <5>;
 		};
 	};
 };
@@ -927,19 +622,48 @@
 &pcie0{
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_pcie0>;
-	disable-gpio = <&gpio1 5 GPIO_ACTIVE_LOW>;
-	reset-gpio = <&gpio4 21 GPIO_ACTIVE_LOW>;
-	ext_osc = <1>;
+	reset-gpio = <&gpio4 20 GPIO_ACTIVE_LOW>;
+	ext_osc = <0>;
 	status = "okay";
 };
 
-&uart1 { /* BT */
+&ecspi2 { /* Wifi 2.4GHz */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_ecspi2>;
+	fsl,spi-num-chipselects = <1>;
+	cs-gpios = <&gpio5 13 GPIO_ACTIVE_LOW>;
+	assigned-clocks = <&clk IMX8MM_CLK_ECSPI2_SRC>;
+	assigned-clock-parents = <&clk IMX8MM_SYS_PLL2_250M>;
+	assigned-clock-rates = <42000000>;
+	/delete-property/ dmas;
+	/delete-property/ dma-names;
+	status = "okay";
+
+	wfx@0 {
+		compatible = "silabs,wfx-spi";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_wfx_irq &pinctrl_wfx_gpios>;
+		interrupts-extended = <&gpio2 11 IRQ_TYPE_EDGE_RISING>;
+		// TODO: leave disabled until we figure out why we get following complaint:
+		//
+		// wfx-spi spi1.0: error while requesting gpio wakeup
+		//
+		// According to docs, driver will simply disable all power saving features when this is not configured.
+		//wakeup-gpios = <&gpio2 10 GPIO_ACTIVE_HIGH>;
+		reset-gpios = <&gpio3 24 GPIO_ACTIVE_HIGH>;
+		reg = <0>;
+		spi-max-frequency = <42000000>;
+	};
+};
+
+&uart1 { /* Zigbee */
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_uart1>;
 	assigned-clocks = <&clk IMX8MM_CLK_UART1_SRC>;
 	assigned-clock-parents = <&clk IMX8MM_SYS_PLL1_80M>;
 	fsl,uart-has-rtscts;
-	resets = <&modem_reset>;
+	reset-gpios = <&gpio4 10 GPIO_ACTIVE_HIGH>, // reset
+					<&gpio3 25 GPIO_ACTIVE_HIGH>; // bootloader
 	status = "okay";
 };
 
@@ -949,41 +673,33 @@
 	status = "okay";
 };
 
-&uart3 {
+&uart3 { /* BlueTooth */
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_uart3>;
 	assigned-clocks = <&clk IMX8MM_CLK_UART3_SRC>;
 	assigned-clock-parents = <&clk IMX8MM_SYS_PLL1_80M>;
 	fsl,uart-has-rtscts;
+	reset-gpios = <&gpio4 11 GPIO_ACTIVE_HIGH>, // reset
+					<&gpio3 19 GPIO_ACTIVE_HIGH>; // bootloader
 	status = "okay";
 };
 
-&usbotg1 {
-	dr_mode = "otg";
-	extcon = <0>, <&typec1_ptn5110>;
-	picophy,pre-emp-curr-control = <3>;
-	picophy,dc-vol-level-adjust = <7>;
+&uart4 { /* Z-Wave */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart4>;
+	assigned-clocks = <&clk IMX8MM_CLK_UART4_SRC>;
+	assigned-clock-parents = <&clk IMX8MM_SYS_PLL1_80M>;
+	reset-gpios = <&gpio4 28 GPIO_ACTIVE_HIGH>; // reset
 	status = "okay";
 };
 
-&usbotg2 {
-	dr_mode = "otg";
-	extcon = <0>, <&typec2_ptn5110>;
-	picophy,pre-emp-curr-control = <3>;
-	picophy,dc-vol-level-adjust = <7>;
-	status = "disabled";
+&usbotg1 { /* USB port */
+	dr_mode = "host";
+	status = "okay";
 };
 
-&usdhc1 {
-	pinctrl-names = "default", "state_100mhz", "state_200mhz";
-	pinctrl-0 = <&pinctrl_usdhc1>, <&pinctrl_usdhc1_gpio>;
-	pinctrl-1 = <&pinctrl_usdhc1_100mhz>, <&pinctrl_usdhc1_gpio>;
-	pinctrl-2 = <&pinctrl_usdhc1_200mhz>, <&pinctrl_usdhc1_gpio>;
-	bus-width = <4>;
-	vmmc-supply = <&reg_sd1_vmmc>;
-	pm-ignore-notify;
-	keep-power-in-suspend;
-	non-removable;
+&usbotg2 { /* LTE modem */
+	dr_mode = "host";
 	status = "okay";
 };
 
@@ -1034,12 +750,3 @@
 &vpu_h1 {
 	status = "okay";
 };
-
-&micfil {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_pdm>;
-	assigned-clocks = <&clk IMX8MM_CLK_PDM_SRC>, <&clk IMX8MM_CLK_PDM_DIV>;
-	assigned-clock-parents = <&clk IMX8MM_AUDIO_PLL1_OUT>;
-	assigned-clock-rates = <0>, <196608000>;
-	status = "okay";
-};
